src/csharp/Grpc.Core/Internal/AsyncCall.cs

```diff
@@ -407,12 +407,15 @@ private void HandleReceivedResponseHeaders(bool success, Metadata responseHeader
         /// <summary>
         /// Handler for unary response completion.
         /// </summary>
-        private void HandleUnaryResponse(bool success, ClientSideStatus receivedStatus, byte[] receivedMessage, Metadata responseHeaders)
+        private void HandleUnaryResponse(bool sucess, ClientSideStatus receivedStatus, byte[] receivedMessage, Metadata responseHeaders)
         {
+            // NOTE: because this event is a result of batch containing GRPC_OP_RECV_STATUS_ON_CLIENT,
+            // success will be always set to true.
```

Comment:
The recv_status_on_client will also make the other ops succeed.
https://github.com/grpc/grpc/blob/master/include/grpc/impl/codegen/grpc_types.h#L307
* This operation always succeeds, meaning ops paired with this operation
      will also appear to succeed, even though they may not have. In that case
      the status will indicate some failure.
*